File: protocol/src/main/java/spp/protocol/probe/error/LiveInstrumentException.java
package spp.protocol.probe.error;

public class LiveInstrumentException extends RuntimeException {

    private final ErrorType errorType;

    public LiveInstrumentException(ErrorType errorType, String message) {
        super(message, null, true, false);
        this.errorType = errorType;
    }

    public LiveInstrumentException toEventBusException() {
        return new LiveInstrumentException(errorType, "EventBusException:LiveInstrumentException[" + errorType + "]: " + getMessage());
    }

    public ErrorType getErrorType() {
        return errorType;
    }

    @Override
    public synchronized Throwable fillInStackTrace() {
        return this;
    }

    @Override
    public String toString() {
        return getLocalizedMessage();
    }

    public enum ErrorType {
        CLASS_NOT_FOUND,
        CONDITIONAL_FAILED
    }
}

File: protocol/src/main/kotlin/spp/protocol/probe/error/LiveInstrumentException.kt
package spp.protocol.probe.error

class LiveInstrumentException(
    val errorType: ErrorType,
    message: String?
) : RuntimeException(message, null, true, false) {

    fun toEventBusException(): LiveInstrumentException {
        return LiveInstrumentException(errorType, "EventBusException:LiveInstrumentException[$errorType]: $message")
    }

    @Synchronized
    override fun fillInStackTrace(): Throwable {
        return this
    }

    override fun toString(): String {
        return localizedMessage
    }

    enum class ErrorType {
        CLASS_NOT_FOUND,
        CONDITIONAL_FAILED
    }
}
